### PR TITLE
chore(Input): use compose for Input

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Changes

Use compose for `Input` and mark `inputRef` as deprecated in favor of `ref` usage

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/test/specs/components/Input/Input-test.tsx [hunk 1]
```diff
@@ -50,6 +51,7 @@ describe('Input', () => {
         consoleUtil.disableOnce();
 
         const wrapper = mount(<Input {...{ [attr]: testValue }} />).find('input');
+
```
Comment (assuncaocharles) [inline packages/fluentui/react-northstar/test/specs/components/Input/Input-test.tsx:54]:
```suggestion
```

File: packages/fluentui/react-northstar/src/themes/teams/components/Input/inputStyles.ts [hunk 2]
```diff
@@ -14,7 +14,6 @@ const inputStyles: ComponentSlotStylesPrepared<InputStylesProps, InputVariables>
     ...(p.error && { border: `${pxToRem(1)} solid ${v.borderColorError}` }),
     ...(p.fluid && { width: '100%' }),
   }),
-
   input: ({ props: p, variables: v }): ICSSInJSStyle => ({
```
Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/themes/teams/components/Input/inputStyles.ts:17]:
```suggestion

  input: ({ props: p, variables: v }): ICSSInJSStyle => ({
```

File: packages/fluentui/react-northstar/src/themes/teams/components/Input/inputStyles.ts [hunk 3]
```diff
@@ -59,7 +59,6 @@ const inputStyles: ComponentSlotStylesPrepared<InputStylesProps, InputVariables>
       padding: p.iconPosition === 'start' ? v.inputPaddingWithIconAtStart : v.inputPaddingWithIconAtEnd,
     }),
   }),
-
   icon: ({ props: p, variables: v }): ICSSInJSStyle => ({
```
Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/themes/teams/components/Input/inputStyles.ts:62]:
```suggestion

  icon: ({ props: p, variables: v }): ICSSInJSStyle => ({
```

File: packages/fluentui/react-northstar/test/specs/components/Input/Input-test.tsx [hunk 4]
```diff
@@ -29,6 +29,7 @@ describe('Input', () => {
       onKeyUp: 'input',
     },
     autoControlledProps: ['value'],
+    forwardsRefTo: `Box[className~="${inputSlotClassNames.input}"]`,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/components/Input/Input-test.tsx:32]:
```suggestion
    forwardsRefTo: 'input',
```

Will this work?

File: packages/fluentui/react-northstar/src/components/Input/Input.tsx [hunk 5]
```diff
@@ -83,7 +74,7 @@ export interface InputProps extends UIComponentProps, ChildrenComponentProps, Su
   /** The HTML input type. */
   type?: string;
 
-  /** Ref for input DOM node. */
+  /** (DEPRECATED) Ref for input DOM node. */
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Input/Input.tsx:77]:
```suggestion
  /** 
    * @deprecated Please use `ref` instead.
    * Ref for input DOM node.
    */
```

**Pull request merged**